scripts/dts/gen_defines.py

```diff
@@ -679,16 +686,22 @@ def write_clocks(node):
                 controller.props["clock-frequency"].val)
 
 
-def str2ident(s):
-    # Converts 's' to a form suitable for (part of) an identifier
+def to_ident(s):
+    # Replaces all non-alphanumeric characters in 's' with underscores, to get
+    # a string suitable for (part of) an identifier.
+    #
+    # Replaces '+' with 'plus' instead of '_'. This is so that to_upper_ident()
+    # generates e.g. DT_COMPAT_ARM_CORTEX_M0PLUS instead of
+    # DT_COMPAT_ARM_CORTEX_M0_ for 'compatible = "arm,cortex-m0+"'.
+
+    return re.sub(r"\W", "_", s.replace("+", "plus"), flags=re.ASCII)
+
+
+def to_upper_ident(s):
```

Comment:
It feels a little weird to have this as a separate function; I immediately wonder how it's different from the more basic `to_ident(s).upper()`.  Saving a character doesn't seem worth it.

Reply:
It's so common that it felt worth it to me. Shaves some bytecode too, though it's not likely to be significant.~